godotenv.go

```diff
@@ -285,23 +292,38 @@ func parseValue(value string, envMap map[string]string) string {
 				case "r":
 					return "\r"
 				default:
-					return c
+					return match
 				}
 			})
+			// unescape characters
+			e := regexp.MustCompile(`\\([^$])`)
+			value = e.ReplaceAllString(value, "$1")
+		}
+
+		if singleQuotes == nil {
+			value = expandVariables(value, envMap)
 		}
 	}
 
-	// expand variables
-	value = os.Expand(value, func(key string) string {
-		if val, ok := envMap[key]; ok {
-			return val
+	return value
+}
+
+func expandVariables(v string, m map[string]string) string {
+	r := regexp.MustCompile(`(\\)?(\$)(\()?\{?([A-Z0-9_]+)?\}?`)
+
+	return r.ReplaceAllStringFunc(v, func(s string) string {
+		submatch := r.FindStringSubmatch(s)
+
+		if submatch == nil {
+			return s
 		}
-		if val, ok := os.LookupEnv(key); ok {
-			return val
+		if submatch[1] == "\\" || submatch[2] == "(" {
```

Comment:
How can the second submatch be `(` shoun't this be `submatch[1] == "\\" || submatch[2] == "("` ? 
But I don't really get the `(\()?` it wouuld make sense for supporting `$(VAR)` but in that case a `(\))?` is missing at the end of the regex and `submatch[2] == "("` isn't needed either.